File: Assets/zes.game.core/scripts/ZES/Lib/Logger.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using UnityEngine;

namespace Zes
{
    public class Logger
    {
        private static Dictionary<string, Logger> loggers = new Dictionary<string, Logger>();

        public static Logger getLogger<T>()
        {
            return getLogger(typeof(T).FullName);
        }

        public static Logger getLogger(string name)
        {
            if (loggers.ContainsKey(name))
            {
                return loggers[name];
            }
            Logger logger = new Logger(name);
            loggers.Add(name, logger);
            return logger;
        }

        private Logger(string name)
        {
            this.name = name;
        }

        private string name;

        private string getMessage(object message)
        {
            return $"[{name}] {message}";
        }

        public void info(object message)
        {
            Debug.Log($"[INF] {getMessage(message)}");
        }

        public void warn(object message)
        {
            Debug.LogWarning($"[WRN] {getMessage(message)}");
        }

        public void error(object message)
        {
            Debug.LogError($"[ERR] {getMessage(message)}");
        }
    }
}

File: Typescripts/src/lib/logger.ts
const loggers = new Map<string, Logger>();

export function getLogger(name: string): Logger {

    let ret = loggers.get(name);
    if (!ret) {
        ret = new Logger(name);
        loggers.set(name, ret);
    }
    return ret;
}

export class Logger {

    constructor(private name: string) { }

    private getMessage(message: string | object) {
        return `[${this.name}] ${message}`;
    }

    debug(message: string | object) {
        console.log(`[debug] ${this.getMessage(message)}`);
    }

    info(message: string | object) {
        console.log(`[info] ${this.getMessage(message)}`);
    }

    warn(message: string | object) {
        console.warn(`[warn] ${this.getMessage(message)}`);
    }

    error(message: string | object) {
        console.error(`[error] ${this.getMessage(message)}`);
    }
}